src/microbe_stage/editor/LightConfigurationPanel.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using Godot;
+
+/// <summary>
+///   The microbe editor's light configuration top panel
+/// </summary>
+public partial class LightConfigurationPanel : PanelContainer
+{
+#pragma warning disable CA2213
+
+    [Export]
+    private Button dayButton = null!;
+
+    [Export]
+    private Button nightButton = null!;
+
+    [Export]
+    private Button averageLightButton = null!;
+
+    [Export]
+    private Button currentLightButton = null!;
+
+#pragma warning disable CA2213
+
+    [Signal]
+    public delegate void OnLightButtonClickEventHandler(string type);
+
+    public enum LightLevelOption
+    {
+        Day,
+        Night,
+        Average,
+        Current,
+    }
+
+    public void OnLightButtonClicked(string type)
+    {
+        EmitSignal(SignalName.OnLightButtonClick, type);
```

Comment:
I think it would be better if this component protected other code from potential string misconfigurations, so I'd like to see the enum LightLevelOption be put in a separate file and this would then translate the GUI data to the C# enum. It might be the case that the event handler needs to use `int` as the type to properly pass the data (as I'm not sure if godot engine detects and supports enums automatically).